lib/grpc_reflection/service/builder/util.ex

```diff
@@ -159,4 +159,54 @@ defmodule GrpcReflection.Service.Builder.Util do
 
   def trim_symbol("." <> symbol), do: symbol
   def trim_symbol(symbol), do: symbol
+
+  def proto_filename(module) do
+    module.__info__(:compile)
+    |> Keyword.get(:source)
+    |> case do
+      nil ->
+        module
+        |> Atom.to_string()
+        |> String.trim_leading("Elixir.")
+        |> then(&(&1 <> ".proto"))
+
+      source ->
+        source
+        |> List.to_string()
+        |> String.trim()
+        |> build_proto_name()
+    end
+  end
+
+  defp build_proto_name(path) do
+    base =
+      cond do
+        String.ends_with?(path, ".pb.ex") -> String.replace_suffix(path, ".pb.ex", "")
+        String.ends_with?(path, ".ex") -> String.replace_suffix(path, ".ex", "")
+        true -> path
+      end
+
+    cond do
+      String.contains?(base, "/priv/protos/") ->
```

Comment:
`/priv/protos/` and `/test/support/protos` are magic strings that might only apply to this project and the examples directories.

It is not appropriate to risk leaking users file directory structure if they don't align with one of these.  This might need some configuration to identify source proto paths to properly and safely trim from dynamic locations